### ISSUE TITLE
Redirections to other domains break persistent client

### ISSUE THREAD

**Issue by tycooon** [httprb/http#557]
```ruby
client = HTTP.persistent("https://payment.euteller.com").follow

**Comment by ixti** [comment]
Yeah, that's unfortunate. Redirection and persistence atm are conflicting when used at the same time. That should be solved by introducing `HTTP::Session` class that will keep pool of persistent-clients per-URI and will properly handle redirections and cookie management.
